etl/snapshot.py

```diff
@@ -353,7 +371,7 @@ def to_table_metadata(self):
 def add_snapshot(
     uri: str,
     filename: Optional[Union[str, Path]] = None,
-    dataframe: Optional[pd.DataFrame] = None,
+    dataframe: Optional[Union[Table, pd.DataFrame]] = None,
     upload: bool = False,
 ) -> None:
```

Comment:
In case you are interested, you can use the following syntax now (since [python 3.10](https://peps.python.org/pep-0604/)):

```python
def add_snapshot(
    uri: str,
    filename: Optional[str | Path] = None,
    dataframe: Optional[Table | pd.DataFrame] = None,
    upload: bool = False,
) -> None:
```

Reply:
Shouldn't we keep compatibility with python 3.9? (At least that's [what we claim](https://github.com/owid/etl)).